### ISSUE TITLE
Error: Raw data option with CURL not supported for PATCH

### ISSUE THREAD

**Issue by ml-eds** [danielstieber/SubscribeToMailchimp#9]
When trying to PATCH an audience member WireHttp throws an error when using CURL:

```
Exception: Raw data option with CURL not supported for PATCH (in /var/www/html/wire/core/WireHttp.php line 643)  
#0 /var/www/html/wire/core/WireHttp.php(484): ProcessWire\WireHttp->sendCURL('https://us18.ap...', 'PATCH', Array) 
#1 /var/www/html/wire/core/Wire.php(389): ProcessWire\WireHttp->___send('https://us18.ap...', '{"language":"de...', 'PATCH') 
#2 /var/www/html/wire/core/WireHooks.php(823): ProcessWire\Wire->_callMethod('___send', Array) 
#3 /var/www/html/wire/core/Wire.php(450): ProcessWire\WireHooks->runHooks(Object(ProcessWire\WireHttp), 'send', Array) 
#4 /var/www/html/site/assets/cache/FileCompiler/site/modules/SubscribeToMailchimp/SubscribeToMailchimp.module(61): ProcessWire\Wire->__call('send', Array) 
#5 /var/www/html/site/assets/cache/FileCompiler/site/templates/_anmeldung_processing.php(73): SubscribeToMailchimp->subscribe('marc.lutter@e-g...', Array, 'a8a63325a8', Array) 
#6 /var/www/html/site/assets/cache/FileCompiler/site/templates/anmeldung.php(1): include('/var/www/html/s...') 
#7 /var/www/htm ....
```

In most current WireHttp version this is the line

https://github.com/processwire/processwire/blob/dev/wire/core/WireHttp.php#L647

According to my tests this can be mitigated by replacing  [line 61](https://github.com/danielstieber/SubscribeToMailchimp/blob/master/SubscribeToMailchimp.module#L61) of current master

```
$response = $http->send($api . '/' . md5($email), json_encode($param), 'PATCH');
```

with

```
$options = array(
    'use' => 'socket',
    'fallback' => 'socket' // false, 'auto', 'socket' or 'curl'
);

$response = $http->send($api . '/' . md5($email), json_encode($param), 'PATCH', $options);
```

I can do a PR, if you want me to.

**Comment by danielstieber** [comment]
@ml-eds thanks for the fix. Yes, PR would be great! Should we do something like 

`protected $httpPatchOptions = array(
        'use' => 'socket',
        'fallback' => 'socket' // false, 'auto', 'socket' or 'curl'
   );`

to reuse it in [line 91](https://github.com/danielstieber/SubscribeToMailchimp/blob/master/SubscribeToMailchimp.module#L91)?

**Comment by danielstieber** [comment]
I made a dev branch for future updates. I've committed 0.0.6 there and added you @ml-eds  as a co-author. Wanna check so I can merge it in?

**Comment by danielstieber** [comment]
@ml-eds Do you want to join as a collaborator? I'm not that active in PW dev currently and it's a bummer that users have to wait for fixes bc of me when you've already had answers ready.

**Comment by ml-eds** [comment]
@danielstieber I am glad to have been helpful with three issues - but I am afraid that I do not have the time to collaborate on a regular basis.

**Comment by danielstieber** [comment]
fixed in v0.0.6
